### ISSUE TITLE
input_ids and labels do not match while using FlaxDataCollatorForT5MLM methods

### ISSUE THREAD

**Issue by alexcpn** [huggingface/transformers#22244]
### System Info

- `transformers` version: 4.27.1
- Platform: Linux-5.18.10-76051810-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 2.0.0.dev20230202+cu116 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@patil-suraj @patrickvonplaten

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

I am following the documentation at    https://huggingface.co/docs/transformers/main/model_doc/t5#training for Unsupervised denoising training  with my dataset
```
    prompt = "The <extra_id_0> walks in <extra_id_1> park"
    encoded_prompt = tokenizer(prompt, truncation=False, padding=False, return_tensors="pt").input_ids
    print(f"encoded_prompt ={encoded_prompt}")
    labels ="<extra_id_0> cute dog <extra_id_1> the <extra_id_2>"
    encoded_labels = tokenizer(labels, truncation=False, padding=False, return_tensors="pt").input_ids
    print(f"encoded_labels ={encoded_labels}")
    print(f"{encoded_prompt.shape} ={encoded_labels.shape}")
```
Output
```
encoded_prompt =tensor([[   37, 32099, 10681,    16, 32098,  2447,     1]])
encoded_labels =tensor([[32099,  5295,  1782, 32098,     8, 32097,     1]])
torch.Size([1, 7]) =torch.Size([1, 7])
````
I am following the script to reproduce the above https://github.com/huggingface/transformers/blob/main/examples/flax/language-modeling/run_t5_mlm_flax.py#L336-L346

```
    prompt = "The cute dog walks in the green park"
    encoded = tokenizer(prompt, truncation=False, padding=False, return_tensors="pt").input_ids
    batch_size =1
    input_length = encoded.shape[1]
    denoiser = FlaxDataCollatorForT5MLM(tokenizer,.35,3)
    mask_indices = np.asarray([denoiser.random_spans_noise_mask(input_length) for i in range(batch_size)])
    labels_mask = ~mask_indices
    input_ids_sentinel = denoiser.create_sentinel_ids(mask_indices.astype(np.int8))
    labels_sentinel = denoiser.create_sentinel_ids(labels_mask.astype(np.int8))
    input_ids = denoiser.filter_input_ids(encoded, input_ids_sentinel)
    labels  =  denoiser.filter_input_ids(encoded, labels_sentinel)
    print(f"input_ids decoded = {tokenizer.decode(*input_ids,skip_special_tokens=False)}")
    print(f"labels decoded   = {tokenizer.decode(*labels,skip_special_tokens=False)}")
    print(f"input_ids.shape {input_ids.shape} should be equal to labels.shape {labels.shape}")
```
This given the denoised output properly, but labels size '(1,5)' is not matching the input size '(1,8)'

```
input_ids decoded = The cute dog walks in the<extra_id_0></s>
labels decoded   = <extra_id_0> green park</s></s>
input_ids.shape (1, 8) should be equal to labels.shape (1, 5)
```

Should I pad the labels with  <extra-ids> to match the size of the input_ids, if not with what should I pad, as the `t5-base` or the  transformer model needs  the input_ids to be the same shape as the labels(targets) for training

Code that can be run https://gist.github.com/alexcpn/b9bb2b0f01833d1bb862502faf99bab8

### Expected behavior

The input_ids and labels should be the same shape

**Comment by alexcpn** [comment]
For t5-training `input_ids` and `labels` need not match, unlike in gpt2 
I was thiking that the denoised training will help it to memorise the text and I guess it kind of does

From https://gist.github.com/alexcpn/e33a8b44e9774653d7492fb494fb1009

```
After Training:'The cute dog walks in the'-->'cute dog walks in the cute cute dog'
```
But the idea in t5 model seems to be to just train it with a specific target (like translation with a prefix) ??
